scala/scala.bzl

```diff
@@ -382,6 +382,13 @@ def collect_srcjars(targets):
             srcjars += [target.srcjars.srcjar]
     return srcjars
 
+def update_jars_to_labels(jars2labels, target, java_provider):
+  for jar in (java_provider.compile_jars + java_provider.transitive_runtime_jars):
+    if jar.path not in jars2labels:
+      if hasattr(target, "jars_to_labels") and jar.path in target.jars_to_labels:
```

Comment:
The bug was that an incorrect target was given in the error message to the user. If we have A, B, and C, (A has B in `deps`, B has C in `deps`) than by the time A was built, the dependency information it receives was already aggregated and attached to the target B.

In order to retrieve information attached to target C, another provider (`jars_to_labels`) was needed to be returned by `scala_libray` in order for targets such as C be exposed when building A